stock_archive_constraint/models/product_product.py

```diff
@@ -3,13 +3,21 @@
 
 from odoo import _, api, models
 from odoo.exceptions import ValidationError
+from odoo.tools import config
 
 
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
+    def _skip_check_archive_constraint_condition(self):
+        return config["test_enable"] and not self.env.context.get(
```

Comment:
@sergio-teruel 

As I suppose you do post-install tests, this could have been avoided throught a configuration parameter that says 'Avoid product archiving when...'.

Such code should be avoided as many as possible

Reply:
Adding a configuration that enables the only goal of the module is also not something optimal. Users expect to install the module to have the only feature it contains without doing anything more, and the problem is only on integration test. This one is a good compromise.

Reply:
I understand as this is a debate I already have many times.

I see two cons:

- Having in code something that depends on tests is not understandable from a pure programmatic point of view
- Activating or deactivating feature through interface is tedious for people that configures Odoo, but has the advantage of not depending on module state.

Reply:
A comment can be added to clarify its usage, although the variable names are very straight-forward, but this is also a compromise solution for reducing the maintenance burden. Having the "interface" option means:

- Add such field in the proper place (company, res.config.settings, system parameter...).
- Add a view that works properly for doing the configuration (res.config.settings is specially ugly).
- Handle translations.
- etc.

It's too much trade-off for a module that does only this feature.

Reply:
> * Add such field in the proper place (company, res.config.settings, system parameter...).
> * Add a view that works properly for doing the configuration (res.config.settings is specially ugly).
> * Handle translations.

Isn't it our normal job ???